cli/flox-rust-sdk/src/models/environment/path_environment.rs

```diff
@@ -401,11 +424,73 @@ impl PathEnvironment {
             Err(EnvironmentError2::WriteEnvJson(e))?;
         }
 
-        // write "run" >> .flox/.gitignore
-        fs::write(dot_flox_path.join(".gitignore"), "run/\n")
-            .map_err(EnvironmentError2::WriteGitignore)?;
+        // write "run" and "cache" to .flox/.gitignore
+        fs::write(dot_flox_path.join(".gitignore"), formatdoc! {"
+            {GCROOTS_DIR_NAME}/
+            {CACHE_DIR_NAME}/
+            "})
+        .map_err(EnvironmentError2::WriteGitignore)?;
+
+        let mut environment = Self::open(pointer, dot_flox_path, temp_dir)?;
+
+        if let Some(InitCustomization {
+            packages: Some(packages),
+            ..
+        }) = customization
+        {
+            // Ignore the result, because we know there can't be packages already installed
+            environment.install(packages, flox)?;
+        }
+
+        Ok(environment)
+    }
+
+    /// Replace all placeholders in the manifest file contents
+    ///
+    /// TODO: we should probably be using toml_edit
+    fn replace_placeholders(
+        contents: &String,
+        system: &str,
+        customization: &Option<InitCustomization>,
+    ) -> String {
+        // Replace system
+        let mut replaced = contents.replace(FLOX_SYSTEM_PLACEHOLDER, system);
+
+        // Don't add example packages if packages are being installed
+        let packages = match customization {
+            Some(InitCustomization {
+                packages: Some(_), ..
```

Comment:
suggestion: add the install descriptors in this function, so that Applying Customization does not need a sequel Applying Customization Part II.

I see we self.install the packages, but we might get away with writing them to the document and then `self.build` the env.
Either way having both of this here seems a bit more self contained

Reply:
I think we should do that once we are using `toml_edit` here, at which point it can call `insert_packages()`. But I don't want to duplicate logic from `insert_packages()` here that we want to drop immediately

Reply:
yes, fine to defer to using a full toml edit approach.